api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java

```diff
@@ -478,8 +478,8 @@ private String getDuplicatePatientsSQLString(List<String> attributes) {
 				innerFields.add("person1." + attribute);
 			} else if (personNameFieldNames.contains(attribute)) {
 				if (!outerSelect.toString().contains("person_name")) {
-					outerSelect.append("inner join person_name t3 on t2.person_id = t3.person_id ");
```

Comment:
Will all the newly introduced tests fail without this change?

Reply:
Not every one of them fails, there are 2 that pass. These are the tests that use "gender" attribute first, so then the `t2` alias is already present in the resulting SQL. However, I think my solution is correct. You can look at other branches of this `if-else` statement, as they all look similar to mine.